nva-datamodel-java/src/main/java/no/unit/nva/model/contexttypes/Publisher.java

```diff
@@ -1,25 +1,34 @@
 package no.unit.nva.model.contexttypes;
 
+import static no.unit.nva.model.contexttypes.utils.MigrateChannelIdUtil.migrateToNewIdIfFound;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import java.net.URI;
 import java.util.Objects;
+import no.unit.nva.model.contexttypes.utils.ChannelType;
 import nva.commons.core.JacocoGenerated;
 
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
 public class Publisher implements PublishingHouse {
+
     private final URI id;
 
     @JsonCreator
     public Publisher(@JsonProperty("id") URI id) {
-        this.id = id;
+        this.id = migrateToNewIdIfFound(id, ChannelType.PUBLISHER);
```

Comment:
A random thought, if an initial check is placed here (and elsewhere this code is called) so that the migration operation occurs only once, we avoid having to remove the code immediately. E.g.:

```
    this.id = checkIfIsNewStyleUri(id) ? id : migrateToNewIdIfFound(id, ChannelType.PUBLISHER);
```

Reply:
Clever thought! Added checkIfIsNewStyleIdentifier() checking if the new identifier is a valid UUID.